### PR TITLE
bluestore: exit early from _dump_transaction() is possible.

### PR DESCRIPTION
**Pull request by rzarzynski**

### CODE REVIEW

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -8314,8 +8314,12 @@ void BlueStore::_dump_extent_map(ExtentMap &em, int log_level)
   }
 }
 
-void BlueStore::_dump_transaction(Transaction *t, int log_level)
+void BlueStore::_dump_transaction(Transaction* const t, const int log_level)
 {
+  if (! cct->_conf->subsys.should_gather(ceph_subsys_bluestore, log_level)) {
```
Comment (rzarzynski) [inline src/os/bluestore/BlueStore.cc]: Unnecessary formatting is quite expensive while the `_dump_transaction` is called from several places, possibly many times per request.

File: src/os/bluestore/BlueStore.cc [hunk 2]
```diff
@@ -8314,8 +8314,12 @@ void BlueStore::_dump_extent_map(ExtentMap &em, int log_level)
   }
 }
 
-void BlueStore::_dump_transaction(Transaction *t, int log_level)
+void BlueStore::_dump_transaction(Transaction* const t, const int log_level)
```
Comment (ifed01) [inline src/os/bluestore/BlueStore.cc]: These const declarations are an overkill IMHO. And they aren't in line with the rest of the bluestore code .

File: src/os/bluestore/BlueStore.cc (hunk 1 above)
Comment (ifed01) [inline src/os/bluestore/BlueStore.cc]: If I understand correctly dout(log_level) is expanded to the following:
#define dout_impl(cct, sub, v)						\
  do {									\
  if (cct->_conf->subsys.should_gather(sub, v)) {			\
    if (0) {								\
      char __array[((v >= -1) && (v <= 200)) ? 0 : -1] __attribute__((unused)); \
    }									\
    static size_t _log_exp_length=80; \
    ceph::logging::Entry *_dout_e = cct->_log->create_entry(v, sub, &_log_exp_length);	\
    ostream _dout_os(&_dout_e->m_streambuf);				\
    CephContext *_dout_cct = cct;					\
    std::ostream* _dout = &_dout_os;

and hence there is no sense in prepending it with and additional check for should_gather.

Comment (liewegas) [inline src/os/bluestore/BlueStore.cc]: Yep, there's an implicit block from dout to dendl; no need to check should_gather explicitly
